core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -225,9 +223,15 @@ protected void setArtificialWayTags(PointList pointList, ReaderWay way) {
         way.removeTag("country");
         way.removeTag("country_rule");
         way.removeTag("custom_areas");
-        List<CustomArea> customAreas = estimatedCenter == null || areaIndex == null
-                ? emptyList()
-                : areaIndex.query(estimatedCenter.lat, estimatedCenter.lon);
+
+        double middleLat = pointList.getLat(pointList.size() / 2), middleLon = pointList.getLon(pointList.size() / 2);
+        List<CustomArea> customAreas;
+        if (areaIndex == null || Double.isNaN(middleLat) || Double.isNaN(middleLon)) {
```

Comment:
Is it actually expected that the coordinates happen to be `NaN`? Otherwise we could just remove this check (and, very minor, not fetch the coordinates at all when `areaIndex==null`?

Reply:
I just followed the existing code to be honest. Calculating the middle point only if the index is present makes sense.

Reply:
Oh, ok. I did not see this. So let's keep it like this (check for NaN) for now.

Reply:
```java
        List<CustomArea> customAreas;
        if (areaIndex != null) {
            double middleLat = pointList.getLat(pointList.size() / 2), middleLon = pointList.getLon(pointList.size() / 2);
            customAreas = areaIndex.query(middleLat, middleLon);
        } else {
            customAreas = emptyList();
        }
```

The WaySegmentParser seems to be the only place where we deliberately create points with NaN coordinates.

Reply:
Done

Reply:
> The WaySegmentParser seems to be the only place where we deliberately create points with NaN coordinates.

Where do you mean exactly?

Reply:
Just checked the codebase if NaN coordinates are a thing and found this:

https://github.com/graphhopper/graphhopper/blob/1e5f4e74f8365bf322bd3ab001afb78e638198e1/core/src/main/java/com/graphhopper/reader/osm/WaySegmentParser.java#L368-L374

Reply:
Ah ok, this can probably be removed anyway, now we merged #2457.

Reply:
I think the reason some points could be missing in theory was that we left out parts of OSM ways when they weren't within a given bounding box that could be specified for the import at some point.

Reply:
I removed this. cb38af67b2d3c0e092babc221d66ceb1c0c6cb4b